### PR TITLE
Referral chart on the main dashboard

### PR DESCRIPTION
**Pull request by yachtcaptain23**
Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.
- [ ] Integrated piwik/matomo (for code that adds new buttons).
- [ ] Addressed or ignored all brakeman warnings

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

**Comment by yachtcaptain23** [comment]
screen shot 2019-03-05 at 9 44 23 am

**Comment by yachtcaptain23** [comment]
Labels:
- [ ] Should be dots rather than a box for the label or time series
- [ ] Would be good to hide year
- [ ] Use dates like March 15 rather than 03-15
- [ ] Swapping colors between 30-day-use
- [ ] Hide gray area

File: app/javascript/publishers/dashboard_chart.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    currentMax = value.retrievals > currentMax ? value.retrievals : currentMax;
+    currentMax = value.first_runs > currentMax ? value.first_runs : currentMax;
+    currentMax = value.finalized > currentMax ? value.finalized : currentMax;
+  });
+  return (currentMax * 100 / 95)
+}
+
+function createChart(data, title, suggestedMax) {
+  var wrapper = document.getElementById('channel-referrals-stats-chart');
+  var canvas = document.getElementById('chanel-referrals-stats-chart-canvas);
```
Comment (corymcdonald) [inline app/javascript/publishers/dashboard_chart.js]: I think you're missing a single quote

File: app/javascript/packs/referral_charts.jsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export function renderReferralCharts() {
+  let props = {
+    referralCodes: JSON.parse(document.getElementById('referrals-hidden-tags').value)
```
Comment (corymcdonald) [inline app/javascript/packs/referral_charts.jsx]: might be better to do something like this

const { value } = document.getElementById('referrals-hidden-tags')
if(value === undefined) {
  return;
}
let referralCodes = JSON.parse(value);

File: app/javascript/packs/referral_charts.jsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
+import * as React from "react";
```
Comment (corymcdonald) [inline app/javascript/packs/referral_charts.jsx]: Can you run prettier on this file?

**Pull request merged**